### DISCUSSION TITLE
Title bar problem test

### DISCUSSION THREAD

**Discussion by Tatsu-syo** [Tatsu-syo/noMeiryoUI#66] [General]
- タイトルバーのフォントサイズに合わせてタイトルバーのサイズを設定するようにしてみた。
- フォントサイズ変更に応じてウインドウサイズを調整するようにした。

----
- Change title bar size using title bar font size.
- Adjust window size using font size settings.

<hr /><em>This discussion was created from the release <a href='https://github.com/Tatsu-syo/noMeiryoUI/releases/tag/TAG-3.1.1_test1'>Title bar problem test</a>.</em>

**Comment by overartem** [comment]
after selecting my custom font, the app froze and closed. The same with Version 3.1.0  (Windows11 PRo 22H2 - 22621.1702)

**Comment by Tatsu-syo** [comment]
If you press set and exit, this behavior is normal.
Some software cause the thread to hang infinitely.
Therefore, the program waits 5 seconds and exits.

**Comment by WinsonDeng99** [comment]
win11标题栏字体还是改不了，辛苦大佬速速更新！

**Comment by ZacharyZhang-NY** [comment]
Win11のタイトルバーのフォントはまだ変更できませんが、アップデートがあることを期待しています！ありがとうございます！

**Comment by lizhengyi3003** [comment]
win11でもタイトルバーは変更できません。タスクバーや多くのグラフでは、タイトルバーの形で表示されています。

**Reply by ZacharyZhang-NY** [reply]
ありがとうございます！

**Comment by Tatsu-syo** [comment]
Hmm, on Windows 11 2022 update, changing title bar font setting causes visual problems.
This is why, I decided to block title bar font change on Windows 11 2022 update.
Application see different font settings.
What I should do?

**Reply by lizhengyi3003** [reply]
視覚的な問題とは何ですか?

**Reply by Tatsu-syo** [reply]
Title bar become ｔransparent. And difficult to see.
Some font avoid problem.
But, everyone don't have such font.

**Reply by lizhengyi3003** [reply]
タイトルバーが透明になったのは、フォント修正の際にレジストリのフォントの地色も初期化してしまったのかもしれません。

**Reply by Tatsu-syo** [reply]
I forgot color settings. Because color settings are separate info font settings.
I have other bug to fix.
I will research and test this problem too.

**Reply by lizhengyi3003** [reply]
タスクバーのフォントは問題ないでしょう。タスクバーのフォントは最初も含めてタイトルバーのフォントを使っているようです。

**Reply by lizhengyi3003** [reply]
。

**Comment by Tatsu-syo** [comment]
I will change title bar font by older version, and check registry related to title bar color.
Whichever it is, it's weird.

**Reply by lizhengyi3003** [reply]
もしあなたは修正ができたら、先に私に送って私にまず試用させて、それから私はあなたに何か修正が必要なところがあるかどうかを見せています。

**Comment by Tatsu-syo** [comment]
I tested to change title bar font on Windows 11H2.
When I changed title bar font, background color didn't change.
But font face also didn't change.
Title bar draw strategy seems to be changed
This is why I can't change title bar font.
Sorry.

**Comment by Tatsu-syo** [comment]
I back out Title bar size using title bar font size.
It confrict to preset WindowMetric settings.
